File: src/main/java/com/demonwav/mcdev/buildsystem/maven/pom/Dependencies.java
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2016 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.buildsystem.maven.pom;

import com.intellij.util.xml.DomElement;
import com.intellij.util.xml.SubTagList;

import java.util.List;

public interface Dependencies extends DomElement {

    @SubTagList("dependency")
    List<Dependency> getDependencies();

    Dependency addDependency();
}

File: src/main/kotlin/com/demonwav/mcdev/buildsystem/maven/pom/Dependencies.kt
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2017 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.buildsystem.maven.pom

import com.intellij.util.xml.DomElement
import com.intellij.util.xml.SubTagList

interface Dependencies : DomElement {

    @get:SubTagList("dependency")
    val dependencies: List<Dependency>

    fun addDependency(): Dependency
}
